execution_workflows/Roar/workflow/Snakefile

```diff
@@ -0,0 +1,124 @@
+"""
+Snakemake pipeline for APA-Scan.
+For help see: https://snakemake.readthedocs.io/en/stable/index.html.
+"""
+
+
+# ------------------------------------------------------------------------------
+# Packages, Libraries and Sources
+
+import pandas as pd
+
+
+
+
+
+# ------------------------------------------------------------------------------
+# Config/Metadata
+
+# Read in config file
+configfile: "config/config.Roar.yaml"
+
+# Convert config to dataframe
+samples = pd.read_csv(os.path.abspath(config["samples"]),sep=",")
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Parameters, filenames & sample data
+
+#### NOTES #####################################################################
+
+# MAY NEED TO ADJUST THE INDEXING DEPENDING ON THE LEVELS, BUT CAN PROBABLY MAKE
+# THIS CUSTOMISABLE FROM THE COMMAND LINE OR A CONFIG FILE ETC. THIS IS BECAUSE
+# THE LAYOUT OF THESE WILL DICTATE THE ORIENTATION OF LEVELS THAT ARE COMPARED
+# BY ROAR. IE FOR THE TEST DATA, THIS MAKES SO SRSF3 is CONDITION1 AND
+# THE CONTROL IS CONDITION 2
+
+################################################################################
+
+
+# Parameters
+THREADS = config["threads"]
+
+
+# Input file patterns
+CONDITION1 = samples["sample_type"].unique()[0]
+CONDITION2 = samples["sample_type"].unique()[1]
```

Comment:
Am I correct in assuming that this will use all samples from either `sample_type` condition? That is, if I have 2 samples in condition1 and 3 samples in condition2, all 5 samples are used for the Roar analysis?

Reply:
I think they are considered as separate groups. The way it is currently set up is to have BAM files for each condition in a separate directory, so this would isolate the two groups. At [lines 15 & 16 in the Roar.R script](https://github.com/iRNA-COSI/APAeval/blob/da00a5b31e730a54b2f0c79adfddfb0cec83c902/execution_workflows/Roar/workflow/scripts/Roar.R#L15) BAM files for each condition are defined separately (saying that I don't like how it's done (they seem to be combined again?), will comment on the actual script).

Roar can only compare between two conditions (I'm almost certain but will double check) so we should also include a check that only two conditions are specified in the sample sheet